publication-rest/src/main/java/no/unit/nva/publication/update/UpdatePublicationHandler.java

```diff
@@ -130,25 +149,149 @@ private static boolean isPending(TicketEntry publishingRequest) {
     }
 
     @Override
-    protected PublicationResponseElevatedUser processInput(UpdatePublicationRequest input,
+    protected PublicationResponseElevatedUser processInput(PublicationRequest input,
                                                            RequestInfo requestInfo,
                                                            Context context)
         throws ApiGatewayException {
+        SortableIdentifier identifierInPath = RequestUtil.getIdentifier(requestInfo);
 
-        var identifierInPath = RequestUtil.getIdentifier(requestInfo);
+        Publication existingPublication = fetchPublication(identifierInPath);
+
+        var userInstance = RequestUtil.createUserInstanceFromRequest(requestInfo, identityServiceClient);
+        var permissionStrategy = PublicationPermissionStrategy.create(existingPublication, userInstance);
+
+        Publication updatedPublication = switch (input) {
+            case UpdatePublicationRequest publicationMetadata ->
+                updateMetadata(publicationMetadata, identifierInPath, existingPublication, permissionStrategy);
+
+            case UnpublishPublicationRequest unpublishPublicationRequest ->
+                unpublishPublication(unpublishPublicationRequest, existingPublication, permissionStrategy);
+
+            case DeletePublicationRequest ignored -> deletePublication(existingPublication, permissionStrategy);
+
+            default -> throw new BadRequestException("Unknown input body type");
+        };
+
+        return PublicationResponseElevatedUser.fromPublication(updatedPublication);
+    }
+
+    private static void throwUnauthorizedUnless(Function<PublicationAction, Boolean> a, PublicationAction requestedPermission) throws UnauthorizedException {
```

Comment:
I suppose this is `authorize()`

Reply:
how about permissionStrategy.authorize(PublicationAction.DELETE) where this was used? then its reusable elsewhere also